tests/node-token.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var vows = require('vows');
+var assert = require('assert');
+var fs = require('fs');
+var events = require('events');
+
+var CodeGen = require('../lib/codegen').CodeGen;
+
+vows.describe('Test Token').addBatch({
+    'Test Automatic Auth Token Assignment': {
+        topic: function(){
+            var swagger = JSON.parse(fs.readFileSync('tests/apis/token.json', 'UTF-8'));
+            /*jshint evil:true*/
+            var Token = eval(CodeGen.getNodeCode({ className: 'Token', swagger: swagger }));
+            return new Token('https://portal.28.io/api');
```

Comment:
Done.
I added the host, basePath and schemas to the swagger spec (protected.json) so after the merge I can update the test to construct without the explicit url.